### ISSUE TITLE
Incorrect fix suggested for FURB189

### ISSUE THREAD

**Issue by penguinolog** [astral-sh/ruff#14274]
Preview check FURB189 suggest using `collections.UserStr`, but correct base class if `collections.UserString` (https://docs.python.org/3/library/collections.html#userstring-objects)

Example code to trigger checker:
```python
class MyStr(str):
    __ slots__ = ()
```

Check command used: `ruff check --preview --select=FURB189`

Ruff version: 0.7.3

**Comment by sbrugman** [comment]
Hi @penguinolog thanks for flagging this!

The issue has already been resolved in main and will be fixed in the upcoming release (https://github.com/astral-sh/ruff/pull/14209).
